src/EditorFeatures/Core/EditAndContinue/EditAndContinueLanguageService.cs

```diff
@@ -80,17 +72,20 @@ public void SetFileLoggingDirectory(string? logDirectory)
         });
     }
 
+    private SolutionServices Services
+        => workspaceProvider.Value.Workspace.Services.SolutionServices;
+
+    private Solution GetCurrentDesignTimeSolution()
+        => workspaceProvider.Value.Workspace.CurrentSolution;
+
     private Solution GetCurrentCompileTimeSolution(Solution? currentDesignTimeSolution = null)
-    {
-        var workspace = WorkspaceProvider.Value.Workspace;
-        return workspace.Services.GetRequiredService<ICompileTimeSolutionProvider>().GetCompileTimeSolution(currentDesignTimeSolution ?? workspace.CurrentSolution);
-    }
+        => Services.GetRequiredService<ICompileTimeSolutionProvider>().GetCompileTimeSolution(currentDesignTimeSolution ?? GetCurrentDesignTimeSolution());
```

Comment:
It's odd that `currentDesignTimeSolution` and `GetCurrentDesignTimeSolution` return different things, but both use the word "Current". Is there a better name for owe of them? `GetCurrentWorkspaceSolution` perhaps? Or is it a bug that `GetCurrentDesignTimeSolution` doesn't return `currentDesignTimeSolution ?? worksp...`?

Reply:
Refactored.